### ISSUE TITLE
add featured image to products if woocommerce is installed?

### ISSUE THREAD

**Issue by jonXmack** [lambdacreatives/tofino#167]
I know we were chatting about featured images earlier but my thought is if we're doing some kind of woocom conditional thing anyway, would it be possible to make featured images appear on products as part of that?

**Comment by danimalweb** [comment]
I don't see why not. Based on my understanding WooComm adds a new post type called: product.

In the same place as all the other WooComm related PHP add the correct Theme Support code to enable support for featured images for this post type.

Does WooComm not have this out of the box? https://docs.woothemes.com/document/featured-images/

**Comment by jonXmack** [comment]
I know I've had to include it manually before, @lesteak had to do it on a job a couple of days ago which is on v1.2 IIRC.

I think if we change `line 20` of `init.php` from `add_theme_support('post-thumbnails', array('post'));` to `add_theme_support('post-thumbnails');` then it'll be up to us to add it in `supports` when setting up custom post types, and can be hidden using the Screen Options tab if it's not needed on pages.

**Comment by danimalweb** [comment]
Would this still enable the default post types ``post`` and ``page`` to have featured images without adding any more code?

In the dev branch right now we have:

```  add_theme_support('post-thumbnails', ['post', 'page']); // Enable featured images for Posts```

**Comment by jonXmack** [comment]
It'll default to true if you don't include the arguments array.

https://codex.wordpress.org/Function_Reference/add_theme_support#Post_Thumbnails

**Comment by danimalweb** [comment]
Nice one. I've just updated the dev branch to remove the arguments array. Cheers for the info.

**Comment by mrchimp** [comment]
I've not had to do this but I probably will at some point. Can I get a tl;dr and/or a note in the docs please?

**Comment by danimalweb** [comment]
TL;DR: All default and future post types will have the featured image option available by default unless specifically turned off using code, or hidden using the Screen Options.
